kratos/geometries/triangle_2d_3.h

```diff
@@ -1876,249 +1878,92 @@ template<class TPointType> class Triangle2D3
+
+        double fex,fey;
         array_1d<double,3 > v0,v1,v2;
-        array_1d<double,3 > axis;
-        array_1d<double,3 > normal, e0, e1 ,e2;
-//
-// 		  /* This is the fastest branch on Sun */
-// 		  /* move everything so that the boxcenter is in (0,0,0) */
+        array_1d<double,3 > e0,e1,e2;
+        std::pair<double, double> min_max;
+
+        /* move everything so that the boxcenter is in (0,0,0) */
         noalias(v0) = triverts[0]- boxcenter;
         noalias(v1) = triverts[1]- boxcenter;
         noalias(v2) = triverts[2]- boxcenter;
-//
-// 		  /* compute triangle edges */
+
+        /* compute triangle edges */
         noalias(e0) = v1 - v0;      /* tri edge 0 */
         noalias(e1) = v2 - v1;      /* tri edge 1 */
         noalias(e2) = v0 - v2;      /* tri edge 2 */
-//
-// 		  /* Bullet 3:  */
-// 		  /*  test the 9 tests first (this was faster) */
-        fex = fabs(e0[0]);
-        fey = fabs(e0[1]);
-        fez = fabs(e0[2]);
-        //AXISTEST_X01(e0[2], e0[1], fez, fey);
-        if (AxisTest_X01(e0[2], e0[1], fez, fey, p0, p2, min,max, rad, v0, v2, boxhalfsize)==0) return false;
-        //AXISTEST_Y02(e0[2], e0[0], fez, fex);
-        if(AxisTest_Y02( e0[2], e0[0], fez, fex, p0, p2, min,max, rad, v0, v2, boxhalfsize)==0) return false;
-        //AXISTEST_Z12(e0[1], e0[0], fey, fex);
-        //if(AxisTest_Z12(e0[1], e0[0], fey, fex, p1, p2, min, max, rad, v1,v2, boxhalfsize )==0) return false;
-
-
-
-        fex = fabs(e1[0]);
-        fey = fabs(e1[1]);
-        fez = fabs(e1[2]);
-        //AXISTEST_X01(e1[2], e1[1], fez, fey);
-        if( AxisTest_X01(e1[2], e1[1], fez, fey, p0, p2, min,max, rad, v0, v2, boxhalfsize)==0) return false;
-        //AXISTEST_Y02(e1[2], e1[0], fez, fex);
-        if(AxisTest_Y02( e1[2], e1[0], fez, fex, p0, p2, min,max, rad, v0, v2, boxhalfsize)==0) return false;
-        //AXISTEST_Z0(e1[1], e1[0], fey, fex);
-        //if( AxisTest_Z0(e1[1], e1[0], fey, fex, p0,  p1, min, max, rad, v0, v1,boxhalfsize)==0) return false;
-
-
-        fex = fabs(e2[0]);
-        fey = fabs(e2[1]);
-        fez = fabs(e2[2]);
-        //AXISTEST_X2(e2[2], e2[1], fez, fey);
-        if (AxisTest_X2(e2[2], e2[1], fez, fey, p0, p1, min, max, rad, v0, v1, boxhalfsize )==0) return false;
-        //AXISTEST_Y1(e2[2], e2[0], fez, fex);
-        if (AxisTest_Y1(e2[2], e2[0], fez, fex, p0, p1, min, max, rad, v0, v1, boxhalfsize )==0) return false;
-        //AXISTEST_Z12(e2[1], e2[0], fey, fex);
-        //if(AxisTest_Z12(e2[1], e2[0], fey, fex, p1, p2, min, max, rad, v1,v2, boxhalfsize ) ==0) return false;
 
+        /* Bullet 3:  */
+        /*  test the 9 tests first (this was faster) */
+        /*  We are only interested on z-axis test, which defines the {x,y} plane */
+        fex = std::abs(e0[0]);
+        fey = std::abs(e0[1]);
+        if (!AxisTest_Z(e0[0],e0[1],fex,fey,v0,v2,boxhalfsize)) return false;
```

Comment:
I will change the comment to explain it better.
The bullet 3 projects the triangle vertices onto a_ij. And then, check the minimum and maximum projection against r. r is the box projection against a_ij as well.
a_ij vector is obtained by a cross product: a_ij = i-axis x j-edge
Since in 2D case, every edge has null z-component, the triangle projections against x,y-axis will allways be 0, and r (x,y-axis) will be 0. So there is no separating axis and the code will continue testing the next check.
In brief, the separating axis can be found performing the z-tests.
This exception is for x,y-axis in bullet 3, the code will perform bullet 2 and the z-axis test.
Bullet 1 is ommited because the box and the triangle are coplanar and there is no separating axis.
I hope you find it useful

Reply:
Sorry, my explanation can be confusing. I have added a comment in the code about the X, Y and Z tests for 2D case

Reply:
So I should check the paper again. Meanwhile is there any test checking the behavior with a box outside of the plane?

Reply:
I have added a test checking a box outside of the plane.
In triangle 2D3, this method does not take care about Z-coordinate. The test returns true because the box overlaps the triangle when everything is projected onto the xy plane